compiler/rustc_errors/src/emitter.rs

```diff
@@ -1839,79 +1834,115 @@ impl EmitterWriter {
                 }
                 row_num += line_end - line_start;
             }
-            for (line_pos, (line, highlight_parts)) in
-                lines.by_ref().zip(highlights).take(MAX_SUGGESTION_HIGHLIGHT_LINES).enumerate()
-            {
-                // Print the span column to avoid confusion
-                buffer.puts(
-                    row_num,
-                    0,
-                    &self.maybe_anonymized(line_start + line_pos),
-                    Style::LineNumber,
-                );
-                if let DisplaySuggestion::Diff = show_code_change {
-                    // Add the line number for both addition and removal to drive the point home.
-                    //
-                    // N - fn foo<A: T>(bar: A) {
-                    // N + fn foo(bar: impl T) {
+            let mut unhighlighted_lines = Vec::new();
+            for (line_pos, (line, highlight_parts)) in lines.by_ref().zip(highlights).enumerate() {
+                debug!(%line_pos, %line, ?highlight_parts);
+
+                let print_line = |line_pos: usize,
+                                  line: &str,
+                                  highlight_parts: &Vec<SubstitutionHighlight>,
+                                  buffer: &mut StyledBuffer,
+                                  row_num: &mut usize| {
```

Comment:
Maybe moving this to its own method might help with readability.

Reply:
I tried, but I'm not sure if it's better, there are a lot of variables / state that needs to be passed around.